### PR TITLE
OSASINFRA-3420: openstack: Decouple OpenStack API calls from Machine generation

### PR DESCRIPTION
**Pull request by pierreprinetti**
To make unit tests possible (and separate responsibilities), remove OSP API calls from the functions that generate Machines and MachineSets.

### CODE REVIEW

**Comment by pierreprinetti** [comment]
cf: https://github.com/openshift/hive/pull/2244

**Comment by 2uasimojo** [comment]
This is great, and will suit hive's needs nicely, thanks @pierreprinetti!

**Comment by 2uasimojo** [comment]
cf. HIVE-2476

**Review by mandre — COMMENTED** [review]
/lgtm

File: pkg/asset/machines/openstack/machines.go [hunk 1]
```diff
@@ -379,10 +373,8 @@ func failureDomainsFromSpec(mpool openstack.MachinePool) []machinev1.OpenStackFa
 	return failureDomains
 }
 
-func checkNetworkExtensionAvailability(cloud, alias string, opts *clientconfig.ClientOpts) (bool, error) {
-	if opts == nil {
-		opts = openstackdefaults.DefaultClientOpts(cloud)
-	}
+func CheckNetworkExtensionAvailability(cloud, alias string) (bool, error) {
```
Comment (mandre) [inline pkg/asset/machines/openstack/machines.go:376]: Linting is not happy about the lack of comment for a public function.

Comment (pierreprinetti) [reply pkg/asset/machines/openstack/machines.go:378]: yup, it's fixed on a force-push, thanks. PTAL

**Comment by 2uasimojo** [comment]
Critical fixes has been turned off; how do we make this go?

**Comment by 2uasimojo** [comment]
Ugh, I'm sorry I didn't notice this on review, but we're going to need to do [this](https://github.com/openshift/installer/pull/4638) again :(

TL;DR: we need to be able to pass custom client opts into CheckNetworkExtensionAvailability(), because we have knobs for custom certs and whatnot.

I'll post the PR.

**Comment by pierreprinetti** [comment]
@2uasimojo Indeed I removed that because the code in Installer only ever passed `nil`. Please add a comment in your PR saying why the function needs to accept connection options, so that we don't remove that again
